### PR TITLE
Array - noble integer

### PR DESCRIPTION
**Pull request by DevratAgarwal**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [ ] Author
    - [ ] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [ ] Author
    - [ ] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [ ] Author
    - [ ] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [ ] Author
    - [ ] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [ ] There are at least 2 test cases
    - [ ] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

File: solutions/arrays/noble-integer.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+let arr=[]
+for(let index=0;index<arraySize;index++)
+{
+	arr.push(parseInt(readLine()));
+}
+let answer=0;
+for(let index=0;index<arraySize;index++)
+{
+	let count=0;
+	for(let index1=0;index1<arraySize;index1++)
```
Comment (shivateja1729) [inline solutions/arrays/noble-integer.js:23]: @DevratAgarwal Lets give an optimal solution in `O(nlogn)` time, I think the students will learn time complexity by the time of arrays.

File: solutions/arrays/noble-integer.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	{
+		if(arr[index1]>arr[index])
+		count=count+1;
+	}
+	if(count===arr[index])
+	answer=answer+1
+}
+if(answer===0)
```
Comment (shivateja1729) [inline solutions/arrays/noble-integer.js:31]: @DevratAgarwal Add a block after every `if`, `else` statement, to make the code look clear.

File: solutions/arrays/noble-integer.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+for(let index=0;index<arraySize;index++)
+{
+	let count=0;
+	for(let index1=0;index1<arraySize;index1++)
+	{
+		if(arr[index1]>arr[index])
+		count=count+1;
+	}
+	if(count===arr[index])
+	answer=answer+1
```
Comment (shivateja1729) [inline solutions/arrays/noble-integer.js:29]: @DevratAgarwal  end every statement with a semicolon `;`.

File: solutions/arrays/noble-integer.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+let fs = require("fs");
+let data = fs.readFileSync(0, 'utf-8');
+let idx = 0;
+data = data.split('\n');
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
```
Comment (shivateja1729) [inline solutions/arrays/noble-integer.js:10]: @DevratAgarwal  PR title should be **Array - Noble integer** as per the PR title naming conventions.

File: solutions/arrays/noble-integer.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+// -------- Do NOT edit anything above this line ----------
+// Use readLine() for taking input, it will read one line of from the input  and is stored in string format
+let arraySize=parseInt(readLine())
+let arr=[]
+for(let index=0;index<arraySize;index++)
+{
+	arr.push(parseInt(readLine()));
+}
+let answer=0;
```
Comment (sailesh-ramayanam) [inline solutions/arrays/noble-integer.js:19]: Initialize this to `-1`.

File: solutions/arrays/noble-integer.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+for(let index=0;index<arraySize;index++)
```
Comment (sailesh-ramayanam) [inline solutions/arrays/noble-integer.js:20]: Please rename `index` to something like `focusedIndex`.

File: solutions/arrays/noble-integer.js (hunk 1 above)
Comment (sailesh-ramayanam) [inline solutions/arrays/noble-integer.js:23]: Please rename `index1` to `currIndex`.

File: solutions/arrays/noble-integer.js (hunk 3 above)
Comment (sailesh-ramayanam) [inline solutions/arrays/noble-integer.js:29]: Once you find the noble integer, set `answer` to `1` and `break` out of the loop.

File: solutions/arrays/noble-integer.js (hunk 2 above)
Comment (sailesh-ramayanam) [inline solutions/arrays/noble-integer.js:31]: With the above suggested changes, you will no longer need a conditional here.

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]
@DevratAgarwal Please address the comments from me and @shivateja1729

File: solutions/arrays/noble-integer.js (hunk 5 above)
Comment (DevratAgarwal) [reply solutions/arrays/noble-integer.js:19]: not needed as i am checking at last if answer if -1 of not and incrementing answer

**Comment by DevratAgarwal** [comment]
@sailesh-ramayanam @shivateja1729 changes have been addressed

**Comment by prathyusha10x** [comment]
@sailesh-ramayanam @shivateja1729  please review the updated code and approve the PR by today EOD

**Review by sailesh-ramayanam — APPROVED** [review]
@DevratAgarwal A few things to follow next time.
1. Please format the code. If you are using VS Code, you can use the built-in formatter. `Right click -> Format Document`
2. Please add comments where necessary
For this PR, I made the changes. Please refer to my commit ( https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions/pull/272/commits/df2dbda32f4377beaadedc5f10675f8df5a91947 ) for the details.

**Review by shivateja1729 — APPROVED** [review]

**Pull request merged**